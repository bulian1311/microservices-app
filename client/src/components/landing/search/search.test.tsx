import React from "react";
import { renderHook } from "@testing-library/react-hooks";
import { render, RenderResult, fireEvent } from "@testing-library/react";
import useStore from "../../../hooks/use-store.hook";
import Search from "./search.component";

let documentBody: RenderResult;
let storeHook = renderHook(() => useStore());

const fetchProductsSpy = jest.spyOn(
  storeHook.result.current.productStore,
  "fetchProducts"
);

const setSearchQuerySpy = jest.spyOn(
  storeHook.result.current.filterStore,
  "setSearchQuery"
);

beforeEach(() => {
  documentBody = render(<Search />);
});

test("Search component match snapshot.", () => {
  const { baseElement } = documentBody;
  expect(baseElement).toMatchSnapshot();
});

test("Search component onChange event.", () => {
  const searchInput = documentBody.getByTestId("search-input");
  fireEvent.change(searchInput, { target: { value: "a" } });

  expect(setSearchQuerySpy).toBeCalled();
  expect(fetchProductsSpy).toBeCalled();
});

test("Search component, click to the cross icon.", () => {
  const crossIcon = documentBody.getByTestId("cross-icon");
  fireEvent.click(crossIcon);

  expect(setSearchQuerySpy).toBeCalled();
  expect(fetchProductsSpy).toBeCalled();
});

test("Search component, click to the example text.", () => {
  const ul = documentBody.getByTestId("example-ul");
  fireEvent.click(ul.firstChild!);

  expect(setSearchQuerySpy).toBeCalled();
  expect(fetchProductsSpy).toBeCalled();
});
