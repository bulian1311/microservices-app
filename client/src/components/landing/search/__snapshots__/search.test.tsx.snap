// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component match snapshot. 1`] = `
<body>
  <div>
    <div
      class="sc-gsTCUz hHNnGB"
    >
      <div
        class="sc-dlfnbm eFBiHn"
      >
        <div
          class="sc-jSgupP iOdEhH"
        >
          <svg
            class="sc-bdfBwQ jCrEt"
            viewBox="0 0 25 25"
            width="1.56"
          >
            <path
              clip-rule="evenodd"
              d="M15.4497 18.005C13.8505 19.1608 11.8937 19.8407 9.7804 19.8407C4.37883 19.8407 0 15.3992 0 9.92035C0 4.44149 4.37883 0 9.7804 0C15.182 0 19.5608 4.44149 19.5608 9.92035C19.5608 12.0639 18.8905 14.0487 17.751 15.6708L24.1771 22.1888C24.8101 22.831 24.8049 23.8584 24.1768 24.4954L24.1499 24.5228C23.5234 25.1582 22.5037 25.1599 21.8758 24.523L15.4497 18.005ZM9.7804 17.5065C13.911 17.5065 17.2595 14.1101 17.2595 9.92035C17.2595 5.73064 13.911 2.3342 9.7804 2.3342C5.64979 2.3342 2.30127 5.73064 2.30127 9.92035C2.30127 14.1101 5.64979 17.5065 9.7804 17.5065Z"
              fill="#4D4D4D"
              fill-opacity="0.8"
              fill-rule="evenodd"
              opacity="0.5"
            />
          </svg>
        </div>
        
        <input
          class="sc-hKgILt cxzGIy"
          data-testid="search-input"
          placeholder="Что вы хотите найти?"
          value=""
        />
        <h2
          class="sc-eCssSg iGyNue"
        >
          Magmer
        </h2>
      </div>
      <div
        class="sc-iBPRYJ craggo"
      >
        например:
        <ul
          class="sc-fubCfw jecXvi"
          data-testid="example-ul"
        >
          <li
            class="sc-pFZIQ sc-jrAGrp iRXoCJ fOYoai"
          >
            детские аттракционы;
          </li>
          <li
            class="sc-pFZIQ iRXoCJ"
          >
            пейнтбол;
          </li>
          <li
            class="sc-pFZIQ iRXoCJ"
          >
            силомер;
          </li>
          <li
            class="sc-pFZIQ iRXoCJ"
          >
            батут
          </li>
        </ul>
      </div>
    </div>
  </div>
</body>
`;
